maploc/data/dataset.py

```diff
@@ -126,44 +127,66 @@ def get_view(self, idx, scene, seq, name, seed, bbox_tile):
         cam_dict = self.data["cameras"][scene][seq][self.data["camera_id"][idx]]
         cam = Camera.from_dict(cam_dict).float()
 
+        # for backward compatibility
         if "roll_pitch_yaw" in self.data:
-            roll, pitch, yaw = self.data["roll_pitch_yaw"][idx].numpy()
+            R_c2w = compose_rotmat(*self.data["roll_pitch_yaw"][idx].numpy())
         else:
-            roll, pitch, yaw = decompose_rotmat(self.data["R_c2w"][idx].numpy())
+            R_c2w = self.data["R_c2w"][idx].numpy()
+
+        t_c2w = self.data["t_c2w"][idx].numpy()
+
         image = read_image(self.image_dirs[scene] / (name + self.image_ext))
 
-        if "plane_params" in self.data:
-            # transform the plane parameters from world to camera frames
-            plane_w = self.data["plane_params"][idx]
-            data["ground_plane"] = torch.cat(
-                [rotmat2d(deg2rad(torch.tensor(yaw))) @ plane_w[:2], plane_w[2:]]
-            )
         if self.cfg.force_camera_height is not None:
             data["camera_height"] = torch.tensor(self.cfg.force_camera_height)
         elif "camera_height" in self.data:
             data["camera_height"] = self.data["height"][idx].clone()
 
         # raster extraction
         canvas = self.tile_managers[scene].query(bbox_tile)
-        xy_w_gt = self.data["t_c2w"][idx][:2].numpy()
-        uv_gt = canvas.to_uv(xy_w_gt)
-        uv_init = canvas.to_uv(bbox_tile.center)
         raster = canvas.raster  # C, H, W
+        world_T_cam = Transform3D.from_Rt(R_c2w, t_c2w).float()
+        world_T_cam2d = Transform2D.camera_2d_from_3d(world_T_cam)
+
+        # gcam: gravity-aligned camera with z=optical axis
+        # gcamxyz: gcam rotated such that z:up,x:right,y:forward.
+
+        gcam_angle = world_T_cam2d.angle - 90
+        Rz = Transform2D.from_degrees(gcam_angle, torch.zeros(2))
+        world_R_gcamxyz = torch.eye(3)
+        world_R_gcamxyz[:2, :2] = Rz.R
+        world_T_gcamxyz = Transform3D.from_Rt(world_R_gcamxyz, world_T_cam.t).float()
+        gcamxyz_T_gcam = Transform3D.from_Rt(
+            Rotation.from_euler("X", -90, degrees=True).as_matrix(), torch.zeros(3)
+        ).float()
+        world_T_gcam = world_T_gcamxyz @ gcamxyz_T_gcam
+        gcam_T_cam = world_T_gcam.inv() @ world_T_cam
+        cam_R_gcam = gcam_T_cam.inv().R
+
+        world_T_tile = Transform2D.from_Rt(torch.eye(2), canvas.bbox.min_).float()
```

Comment:
You need to be very careful with manipulating world (ENU) coordinates: if these are very large (large area), using fp32 might round them significantly. Maybe check this and potentially use double until the coordinates are computed w.r.t. the tile origin?

Reply:
Thanks for pointing that out. I checked how large the bboxes are for 3 cities from MGL and calculated the precision fp32 would give us for this range. The bounds of these cities are: 
```
sanfrancisco_soma: -1588.92329738242,-1114.3897402399052,966.4837601301315,2156.016168089973
amsterdam: -3037.752837387475,-2846.298378785683,2564.613987351439,3050.530929210985
berlin: -2291.9631479936756,-2552.234173679477,2113.2678937757055,2429.1671109630406
```
The maximum absolute value is 3050.530929210985. Casting absolute values as high as 4096 to fp32 would round them off to the nearest 1/2^12 m  (0.00024 m), so the max possible rounding error is half of that, ~0.00012 m. Assuming the rest of the cities are not much bigger than these 3, this rounding error would be insignificant, right?

Anyway, I think it's still safer to stick to fp64 for world coordinates, so I've fixed it here: 8d11e77

Reply:
You're right, thanks for checking!!